lib/transloadit/assembly.rb

```diff
@@ -153,4 +158,22 @@ def _wrap_steps_in_hash(steps)
   def _extract_options!(args)
     args.last.is_a?(Hash) ? args.pop : {}
   end
+
+  #
+  # Stays idle for certain time and then reopens assembly files for reprocessing.
+  # Should be called when assembly rate limit is reached.
+  #
+  # @param  [Response] response  assembly response that comes with a rate limit
+  # @param [Array<IO>] ios the files sent for the assembly to process.
+  #
+  def _handle_rate_limit!(response, ios, is_retrying)
+    if is_retrying
+      warn "Rate limit reached. Waiting for #{response.wait_time} seconds before retrying."
+      sleep response.wait_time
+      # reopen file stream
```

Comment:
I'd rather write *why* we reopen the file stream, because the code already says what it does :)